src/rule-unused-fragments.js

```diff
@@ -0,0 +1,136 @@
+  const fragmentNames = {};
+
+  function walkAST(node) {
+    if (node.kind === 'FragmentSpread') {
+      const nameNode = node.name;
+      fragmentNames[nameNode.value] = nameNode;
+    }
+    if (node.kind === 'OperationDefinition') {
+      if (node.operation === 'mutation' || node.operation === 'subscription') {
+        return;
+      }
+      node.selectionSet.selections.forEach(selection => {
+        walkAST(selection);
+      });
+      return;
+    }
+    for (const prop in node) {
+      const value = node[prop];
+      if (prop === 'loc') {
+        continue;
+      }
+      if (value && typeof value === 'object') {
+        walkAST(value);
+      } else if (Array.isArray(value)) {
+        value.forEach(child => {
+          walkAST(child);
+        });
+      }
+    }
+  }
+
+  walkAST(graphQLAst);
+  return fragmentNames;
+}
+
+function getGraphQLFragmentDefinitionName(graphQLAst) {
+  const fragmentDefinition = graphQLAst.definitions.find(
+    definition => definition.kind === 'FragmentDefinition'
+  );
+  if (fragmentDefinition) {
+    return fragmentDefinition.name.value;
+  } else {
+    return null;
+  }
+}
+
+function isGraphQLTemplate(node) {
+  return (
+    node.tag.type === 'Identifier' &&
+    node.tag.name === 'graphql' &&
+    node.quasi.quasis.length === 1
+  );
+}
+
+function rule(context) {
+  let foundImportedModules = [];
+  let templateLiterals = [];
+
+  return {
+    Program(_node) {
+      foundImportedModules = [];
+      templateLiterals = [];
+    },
+    'Program:exit'(_node) {
+      const fragmentsInTheSameModule = [];
+      templateLiterals.forEach(templateLiteral => {
+        const graphQLAst = getGraphQLAST(templateLiteral);
+        if (!graphQLAst) {
+          // ignore nodes with syntax errors, they're handled by rule-graphql-syntax
+          return;
+        }
+        const fragmentName = getGraphQLFragmentDefinitionName(graphQLAst);
+        if (fragmentName) {
+          fragmentsInTheSameModule.push(fragmentName);
+        }
+      });
+
+      templateLiterals.forEach(templateLiteral => {
+        const graphQLAst = getGraphQLAST(templateLiteral);
+        if (!graphQLAst) {
+          // ignore nodes with syntax errors, they're handled by rule-graphql-syntax
+          return;
+        }
+
+        const queriedFragments = getGraphQLFragmentNames(graphQLAst);
+        for (const fragment in queriedFragments) {
+          const matchedModuleName = foundImportedModules.find(name =>
+            fragment.startsWith(name)
+          );
+          if (
+            !matchedModuleName &&
+            !fragmentsInTheSameModule.includes(fragment)
+          ) {
+            context.report({
+              node: templateLiteral,
+              loc: utils.getLoc(
+                context,
+                templateLiteral,
+                queriedFragments[fragment]
+              ),
+              message:
+                `This queries for the fragment \`${fragment}\` but this file does ` +
```

Comment:
Since we're renaming this rule to `must-colocate-fragment-spreads`, could you update the error message as well?

```
`This spreads the fragment \`${fragment}\` but ` +
'this module does not use it directly. If a different module ' +
'needs this information, that module should directly define a ' +
'fragment querying for that data, colocated next to where the ' +
'data is used.',
```